### DISCUSSION TITLE
Error on mock modules

### DISCUSSION THREAD

**Discussion by konpeki622** [vitest-dev/vitest#695] [Q&A]
I used module mock like this. It has no problem in jest but caused error when with vitest(0.2.8). Can you help me?
```js
vi.mock('vite', () => {
  const server = {
    listen: vi.fn(),
  }
  return {
    createServer: vi.fn().mockReturnValue(server),
  }
})

// another file
const server= await createServer(config)
server.listen()
```
error message: `TypeError: Cannot read properties of undefined (reading 'listen')`

**Comment by sheremet-va** [comment]
Can you give a proper reproduction? The reason can be anything 👀

**Reply by konpeki622** [reply]
Thank you. I resolved it. I should set environment to jsdom.
